src/main/java/frc/robot/RobotContainer.java

```diff
@@ -42,6 +43,7 @@
 
   // Controllers
   private XboxController controlXbox = new XboxController(0);
+  private JoystickButton xboxAButton = new JoystickButton(controlXbox, XboxController.Button.kA.value);

```

Comment:
why not use the controller directly, is that not possible

Reply:
https://docs.wpilib.org/en/stable/docs/software/commandbased/binding-commands-to-triggers.html
https://first.wpi.edu/FRC/roborio/beta/docs/java/edu/wpi/first/wpilibj/XboxController.html#getAButtonPressed--
I think the `JoystickButton` class just wraps the `XboxController`'s button methods so you can do event based "when pressed" stuff instead of having to check periodically whether the button was pressed using `getAButtonPressed()` and then going from there. Je crois que setting up button bindings via `JoystickButton` is the reason for the `controllerBindings` method.